homeassistant/components/foscam/entity.py

```diff
@@ -0,0 +1,36 @@
+"""Component providing basic support for Foscam IP cameras."""
+from __future__ import annotations
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity import Entity
+
+from .const import DOMAIN
+from .coordinator import FoscamCoordinator
+
+
+class FoscamEntity(Entity):
+    """Base entity for Foscam camera."""
+
+    def __init__(
+        self,
+        coordinator: FoscamCoordinator,
+        config_entry: ConfigEntry,
+    ) -> None:
+        """Initialize the base Foscam entity."""
+
+        self._username = config_entry.data[CONF_USERNAME]
+        self._password = config_entry.data[CONF_PASSWORD]
+
+        dev_info = coordinator.data.get("dev_info", None)
+
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, config_entry.entry_id)},
+            manufacturer="Foscam",
+            name=config_entry.title,
```

Comment:
Can also be removed

Reply:
why can this be removed? that is what will be common between multiple entities

Reply:
A primary device without name will use the config entry title by default, so removing does the same

Reply:
Ok, sorry I thought you were pointing at the full _attr_device_info. Removed it